charts/datadog/templates/_container-agent.yaml

```diff
@@ -142,6 +142,11 @@
     {{- if eq .Values.targetSystem "linux" }}
     - name: dsdsocket
       mountPath: {{ (dir .Values.datadog.dogstatsd.socketPath) }}
+    {{- if .Values.providers.eks.ec2.useHostnameFromFile }}
+    - name: cloudinit-instance-id-file
+      mountPath: /var/lib/cloud/data/instance-id
+      readOnly: true
+    {{- end }}
```

Comment:
to avoid duplicating the same logic in each containers, maybe you can create a reusable "template" like the one present in https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_container-cri-volumemounts.yaml